tests/go.mod

```diff
@@ -13,7 +12,16 @@ require (
 
 require (
 	github.com/accuknox/auto-policy-discovery/src v0.0.0-20230320081550-86bcb8b555bd
+	github.com/kubearmor/KubeArmor/pkg/KubeArmorHostPolicy v0.0.0-20220823050108-4455a183e9ef
+	github.com/kubearmor/KubeArmor/pkg/KubeArmorPolicy v0.0.0-20220823050108-4455a183e9ef
+	github.com/kubearmor/KubeArmor/protobuf v0.0.0-20220815044951-425f333210e1
+	github.com/kubearmor/kubearmor-client v0.8.3
```

Comment:
This defeats the point of the change. The change was required to remove dependency on `github.com/kubearmor/KubeArmor/tests` by having the utils function in the repo. 

For this to be usable with the latest versions of `k8s.io...`, we still need to update the packages starting with `github.com/kubearmor/KubeArmor/...` because they are using older versions. (See https://github.com/kubearmor/KubeArmor/pull/1207)

Ideally, we should not depend on these packages, and all util methods can migrate to the dynamic client instead of the typed client for CRUD operations with resources.

---

The whole issue is that for updating dependency versions in DE, we need to update test dependencies too as it relies on the `src` folder.

`KubeArmor` imports were hindering updation of `test` dependencies, hence it was required to remove dependency from `KubeArmor` main repo to perform tests.

Reply:
> KubeArmor imports were hindering updation of test dependencies, hence it was required to remove dependency from KubeArmor main repo to perform tests.

@Vyom-Yadav are you suggesting to remove all code dependency from kubearmor mainrepo ?

Reply:
> @Vyom-Yadav are you suggesting to remove all code dependency from kubearmor mainrepo ?

Yes, at least in the test module where we only use utils from Kuberarmor test module.

> Edit: Should we rewrite all the CRUD operation to use dynamic client?

You can use typed client for normal k8s resources. You can use the dynamic client for CRDs. We can keep the rest of it, just want to remove dependency from KubeArmor in test module.